sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -80,19 +81,46 @@ public Response Response
         /// </summary>
         public TimeSpan? NetworkTimeout { get; set; }
 
-        internal void AddPolicies(RequestContext? context)
+        internal void ApplyRequestContext(RequestContext? context)
```

Comment:
This routine works only if it's called once. It might be good to add a Debug.Assert to ensure it's not called more than once, or change the error code array assignments to logic that can resize the arrays if they already exist.

Reply:
Moved away from the array approach.